node/peers/peers.go

```diff
@@ -328,7 +328,14 @@ func (pm *PeerMan) maintainMinPeers(ctx context.Context) {
 				if !bk.try() {
 					if bk.maxedOut() {
 						pm.log.Warnf("Failed to connect to peer %s (%v) after %d attempts", peerIDStringer(pid), pid, bk.attempts)
-						pm.removePeer(pid)
+
+						// Auto-blacklist peer that exhausted connection retries if enabled
+						if pm.blacklistConfig.Enable && pm.blacklistConfig.AutoBlacklistOnMaxRetries {
+							pm.log.Warnf("Auto-blacklisting peer %s due to connection exhaustion after %d attempts", peerIDStringer(pid), bk.attempts)
+							pm.BlacklistPeer(pid, "connection_exhaustion", 0) // permanent blacklist (also removes peer)
```

Comment:
not sure if I agree that it should be permanent... thinking about situations where a peer is just disconnected for bad network, etc. maybe better to have something like 1 hours or something

but we should also stay wary after we implement that to be sure this doesn't happen very frequently with fine nodes